controller/AppointmentController.ts

```diff
@@ -0,0 +1,86 @@
+import { NextFunction, Request, Response } from "express";
+import BookingService from "../service/BookingService";
+import LabService from "../service/LabService";
+import PatientService from "../service/PatientService";
+import UserService from "../service/UserService";
+
+export interface IAppointmentController {
+  appointments: (req: Request, res: Response, next: NextFunction) => {};
+}
+
+interface AppointmentDTO {
+    id: string,
+    lab: {
+        name: string,
+        price: string,
+    },
+    patient: {
+        name: string,
+        blood_group: string,
+        address: string,
+    },
+    booking: {
+        name_of_payee: string,
+        payment: string,
+    },
+    date: string,
+    timeslot: number,
+    status: number,
+}
+
+function AppointmentController(
+  bookingService: BookingService,
+  userService: UserService,
+  labService: LabService,
+  patientService: PatientService,
+) {
+  const appointments = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    const userID = res.locals.userID;
+    try {
+      const data = await bookingService.getUserAppointments(userID);
+      const tempData = await Promise.all(data.map(async (value) => {
+        const userData = await userService.getById(value.user);
+        const labData = await labService.getById(value.lab);
+        const patientData = await patientService.getById(value.patient);
+
+        return {
+          id: value._id,
+          lab: {
+              name: labData.name,
+              price: labData.price
+          },
+          patient: {
+              name: patientData.name,
```

Comment:
you can destruct the object in line number 48 and you can directly pass the variables, ex.patient:{ name, blood_group, address}. You are passing same name variables so destructing it and passing variables directly is good practice. This also be good practise for the sake of practice intimating.